test/com/esotericsoftware/kryo/serializers/GenericsTest.java

```diff
@@ -129,6 +135,47 @@ public void testNotPassingToSuper () {
 		kryo.register(DontPassToSuper.class);
 		kryo.copy(new DontPassToSuper());
 	}
+	
+	
+
+	@Test
+	public void testComplicatedGenerics() {
```

Comment:
I have replaced this test-case with a more concise version in #717. Please merge the current master into your branch and verify that all three tests from #717 pass.

Reply:
I've pulled from the master branch and reverted my changes on that file. The tests pass.